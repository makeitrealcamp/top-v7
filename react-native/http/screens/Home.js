import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

export default function Home({ navigation }) {
  async function handleLogin() {
    const token = 'asdfawerasdrf'
    await AsyncStorage.setItem('token', token)
  }

  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <Button
        title="login"
        onPress={handleLogin}
      />
      <Button
        title="posts"
        onPress={() => navigation.navigate('Posts')}
      />
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
