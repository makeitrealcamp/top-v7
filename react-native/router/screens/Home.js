import React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, Button } from 'react-native';

export function Home({ navigation }) {
  return (
    <View style={styles.container}>
      <Text>Home</Text>
      <Button
        title="Go to About"
        onPress={() => navigation.navigate('About', {
          params1: 'HOLA',
          params2: 5
        })}
      />
      <StatusBar style="auto" />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
