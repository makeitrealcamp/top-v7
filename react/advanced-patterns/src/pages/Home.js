import Title from '../components/Title'

function Home(props) {
  console.log('Home', props);
  return <Title>Home</Title>
}

export default Home
