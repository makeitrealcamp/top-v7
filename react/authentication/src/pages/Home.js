export function Home() {
  return <h1>Hola mundo</h1>
}
