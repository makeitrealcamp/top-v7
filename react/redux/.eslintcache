[{"/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/App.js":"1","/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/components/Counter.js":"2","/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/store/index.js":"3","/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/components/Button.js":"4"},{"size":790,"mtime":1606144990074,"results":"5","hashOfConfig":"6"},{"size":608,"mtime":1606141301056,"results":"7","hashOfConfig":"6"},{"size":651,"mtime":1606142937277,"results":"8","hashOfConfig":"6"},{"size":463,"mtime":1606142229468,"results":"9","hashOfConfig":"6"},{"filePath":"10","messages":"11","errorCount":0,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0,"source":null},"zs6cow",{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"14","messages":"15","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"16","messages":"17","errorCount":0,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0,"source":null},"/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/App.js",["18"],"/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/components/Counter.js",[],"/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/store/index.js",[],"/Users/simonhoyos/Documents/makeitreal/top/top-v7/react/redux/src/components/Button.js",["19"],{"ruleId":"20","severity":1,"message":"21","line":4,"column":10,"nodeType":"22","messageId":"23","endLine":4,"endColumn":19},{"ruleId":"20","severity":1,"message":"24","line":1,"column":10,"nodeType":"22","messageId":"23","endLine":1,"endColumn":17},"no-unused-vars","'INCREMENT' is defined but never used.","Identifier","unusedVar","'connect' is defined but never used."]