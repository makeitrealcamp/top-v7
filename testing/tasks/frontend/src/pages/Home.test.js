import React from 'react'
import { render, cleanup } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import Home from './Home'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Home', () => {
  afterEach(() => {
    cleanup()
  })

  it('should match snapshot', () => {
    const store = mockStore({
      tasksReducer: {
        tasks: []
      }
    })

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
