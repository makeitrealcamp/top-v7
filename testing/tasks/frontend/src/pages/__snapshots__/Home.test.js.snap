// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    class="tasks"
  >
    <h1>
      Tasks
    </h1>
    <p>
      No tasks created yet. Create one 
      <a
        href="/create"
      >
        here
      </a>
    </p>
  </div>
</div>
`;
